indexer/validate_and_format_contacts.cpp

```diff
@@ -403,7 +444,70 @@ bool ValidateLinePage(string const & page)
 
   string const domain = strings::MakeLowerCase(url::Url::FromString(page).GetHost());
   // Check Line domain name.
-  return (domain == "line.me" || strings::EndsWith(domain, ".line.me"));
+  return (domain == dLine || strings::EndsWith(domain, dDotLine));
+}
+
+bool isSocialContactTag(string_view tag)
+{
+  return tag == kInstagram || tag == kFacebook || tag == kTwitter || tag == kVk || tag == kLine;
+}
+
+bool isSocialContactTag(MapObject::MetadataID const metaID)
+{
+  return metaID == MapObject::MetadataID::FMD_CONTACT_INSTAGRAM ||
+         metaID == MapObject::MetadataID::FMD_CONTACT_FACEBOOK ||
+         metaID == MapObject::MetadataID::FMD_CONTACT_TWITTER ||
+         metaID == MapObject::MetadataID::FMD_CONTACT_VK ||
+         metaID == MapObject::MetadataID::FMD_CONTACT_LINE;
+}
+
+// Functions ValidateAndFormat_{facebook,instagram,twitter,vk}(...) by default strip domain name
+// from OSM data and user input. This function prepends domain name to generate full URL.
+string socialContactToURL(string_view tag, string_view value)
+{
+  ASSERT(!value.empty(), ());
+
+  if (tag == kInstagram)
+    return string{urlInstagram} + string{value};
```

Comment:
It is not critical, but it is better to avoid unnecessary memory allocation for a temporary string object. The optimal way to do it (here and in other places):
```suggestion
    return string{urlInstagram}.append(value);
```

Reply:
Replaced `+` with `.append(...)` in the file.